lang/syntax.md

```diff
@@ -93,10 +93,13 @@ enum Terminator {
 And finally, the syntax of expressions:
 
 ```rust
+/// Constants are Values, but are not allowed to be Value::Ptr | Value::Union.
```

Comment:
The comment is not accurate -- they can still be a pointer with `None` provenance.

Reply:
```suggestion
/// Constants are Values, but cannot have provenance.
/// (We actually entirely disallow Value::Ptr | Value::Union in the well-formedness checker.)
```